### DISCUSSION TITLE
Init inside interactive Fish or outside?

### DISCUSSION THREAD

**Discussion by MLSci** [fish-shell/fish-shell#10914] [Q&A] [answered]
My default `config.fish` is
```
if status is-interactive
    # Commands to run in interactive sessions can go here
end
```
Should I put `starship init fish | source` inside or outside the `if...end`?

**Comment by faho** [comment, marked as answer]
starship is a prompt. That means it is the thing that is shown when a command has completed and fish is waiting for input.

That *is* interactive - like that's the core thing that interactive mode does. So it only makes sense to enable starship when the shell is interactive.
